pkgs/applications/emulators/darling/default.nix

```
@@ -0,0 +1,172 @@
+{ clangStdenv
+, lib
+, runCommandWith
+, writeShellScript
+, fetchFromGitHub
+, fetchpatch
+
+, freetype
+, libjpeg
+, libpng
+, libtiff
+, giflib
+, xorg
+, cairo
+, libglvnd
+, fontconfig
+, dbus
+, libGLU
+, fuse
+, ffmpeg
+, pulseaudio
+
+, python2
+, python3
+, cmake
+, ninja
+, pkg-config
+, bison
+, flex
+
+, libbsd
+, openssl
+
+, addOpenGLRunpath
+}:
+let
+  stdenv = clangStdenv;
+
+  # The build system invokes clang to compile Darwin executables.
+  # In this case, our cc-wrapper must not be used.
+  ccWrapperBypass = runCommandWith {
+    inherit stdenv;
+    name = "cc-wrapper-bypass";
+    runLocal = false;
+    derivationArgs = {
+      template = writeShellScript "template" ''
+        for (( i=1; i<=$#; i++)); do
+          j=$((i+1))
+          if [[ "''${!i}" == "-target" && "''${!j}" == *"darwin"* ]]; then
+            # their flags must take precedence
+            exec @unwrapped@ "$@" $NIX_CFLAGS_COMPILE
+          fi
+        done
+        exec @wrapped@ "$@"
+      '';
+    };
+  } ''
+    unwrapped_bin=${stdenv.cc.cc}/bin
+    wrapped_bin=${stdenv.cc}/bin
+
+    mkdir -p $out/bin
+
+    unwrapped=$unwrapped_bin/$CC wrapped=$wrapped_bin/$CC \
+      substituteAll $template $out/bin/$CC
+    unwrapped=$unwrapped_bin/$CXX wrapped=$wrapped_bin/$CXX \
+      substituteAll $template $out/bin/$CXX
+
+    chmod +x $out/bin/$CC $out/bin/$CXX
+  '';
+
+  wrappedLibs = [
+    # src/native/CMakeLists.txt
+    freetype
+    libjpeg
+    libpng
+    libtiff
+    giflib
+    xorg.libX11
+    xorg.libXext
+    xorg.libXrandr
+    xorg.libXcursor
+    xorg.libxkbfile
+    cairo
+    libglvnd
+    fontconfig
+    dbus
+    libGLU
+
+    # darling-dmg
+    fuse
+
+    # CoreAudio
+    ffmpeg
+    pulseaudio
+  ];
+in stdenv.mkDerivation {
+  pname = "darling";
+  version = "unstable-2023-04-21";
+
+  src = fetchFromGitHub {
+    owner = "darlinghq";
+    repo = "darling";
+    rev = "759f5479a669d77451dba6267de3fd1ca253eaac";
+    fetchSubmodules = true;
+    hash = "sha256-hHow0zAxVWMh8gGhctQwolpw2ARIspIrLhNO2ibktRQ=";
+  };
+
+  patches = [
+    # https://github.com/darlinghq/darling/pull/1355
+    (fetchpatch {
+      url = "https://github.com/darlinghq/darling/commit/8fa8c70d7db7db4be140ecb0758f78d13e72de2a.patch";
+      hash = "sha256-gxXeNNutvEplpX8Wj2Sd17BxBWds+fmYo4vxs2Ip7vk=";
+    })
+  ];
+
+  postPatch = ''
+    chmod +x src/external/bootstrap_cmds/migcom.tproj/mig.sh
+    patchShebangs .
+
+    sed -i 's|SETUID||g' src/startup/CMakeLists.txt
+    sed -i 's|SETGID||g' src/external/basic_cmds/CMakeLists.txt
```

```suggestion
    substituteInPlace src/startup/CMakeLists.txt --replace SETUID ""
    substituteInPlace src/external/basic_cmds/CMakeLists.txt --replace SETGID ""
```

`substituteInPlace` warns when the pattern is not found.